tests/platform_tests/api/test_chassis.py

```diff
@@ -378,8 +380,7 @@ def test_sfps(self, duthosts, enum_rand_one_per_hwsku_hostname, localhost, platf
             num_sfps = int(chassis.get_num_sfps(platform_api_conn))
         except:
             pytest.fail("num_sfps is not an integer")
-        list_sfps = []
-        list_sfps = physical_port_indices
+        list_sfps = [physical_port_indices[intf] for intf in natsorted(physical_port_indices.keys())]
```

Comment:
@blueve could fix the fixture 'physical_port_indices' at line 64 instead of here.? Also, handle a case for duplicate indices when the port is a breakout port. You can use set().

Reply:
I am worry about the `set()` will not respect the ordering. Let use a separate set

Reply:
you can take a look here [https://github.com/Azure/sonic-mgmt/pull/4244](https://github.com/Azure/sonic-mgmt/pull/4244) the indices are sorted and duplicate ones removed.

Reply:
@prgeor Thanks, I checked in SO: https://stackoverflow.com/questions/15800296/does-converting-a-list-to-a-set-change-the-order-of-elements

Seems like convert list to set might change the elements' ordering. Since we depends on the ordering, I updated the PR with a safer method. I recommend you to use similar way.